diskqueue.go

```diff
@@ -425,6 +634,24 @@ func (d *diskQueue) writeOne(data []byte) error {
 		return fmt.Errorf("invalid message write size (%d) minMsgSize=%d maxMsgSize=%d", dataLen, d.minMsgSize, d.maxMsgSize)
 	}
 
+	totalBytes := int64(4 + dataLen)
+	reachedFileSizeLimit := false
+
+	if d.enableDiskLimitation {
+		// check if we will reach or surpass file size limit
+		if d.writePos+totalBytes+numFileMsgBytes >= d.maxBytesPerFile {
+			reachedFileSizeLimit = true
+		}
+
+		// free disk space if needed
+		err = d.checkDiskSpace(totalBytes, reachedFileSizeLimit)
+		if err != nil {
+			return err
+		}
+	} else if d.writePos+totalBytes >= d.maxBytesPerFile {
+		reachedFileSizeLimit = true
+	}
```

Comment:
nit: reduce changes to case where disklimit is not enabled. In this way, all new changes are in `if d.enableDiskLimitation`.
```go
	if d.enableDiskLimitation {
		reachedFileSizeLimit := false
		// check if we will reach or surpass file size limit
		if d.writePos+totalBytes+numFileMsgBytes >= d.maxBytesPerFile {
			reachedFileSizeLimit = true
		}

		// free disk space if needed
		err = d.checkDiskSpace(totalBytes, reachedFileSizeLimit)
		if err != nil {
			return err
		}
	}
```